cmake/modules/openmp.cmake

```diff
@@ -4,21 +4,40 @@ option(USE_OpenMP "Use OpenMP" ON)
 # This finds the library path from the system's clang for OpenMP
 #
 # On Fedora, it's at the same place as others, so we don't need to look elsewhere
-# On Ubuntu, it's in /usr/lib/llvm-${version}, so find_package finds GOMP for GCC instead.
-execute_process (
-    COMMAND bash -c "for lib in $(clang -lomp -### 2>&1); do echo $lib | grep -o \"\\/.*llvm.*\\w\"; done"
-    OUTPUT_VARIABLE LLVM_OMP_PATH
-)
-# Only if we found an "llvm" path that we need to add
-if (LLVM_OMP_PATH)
-  set(CMAKE_PREFIX_PATH ${LLVM_OMP_PATH})
-endif()
+# On Ubuntu, it's in /usr/lib/llvm-${version}, so GCC finds GOMP instead.
 
 if(USE_OpenMP)
-  find_package(OpenMP)
-  if(OPENMP_FOUND)
+  # Only bypass for GCC
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # Only if we found an "llvm" path that we need to add
+    find_library(LLVM_OMP NAMES omp libomp libomp.so libomp.so.5)
+    find_library(PTHREAD NAMES pthread libpthread libpthread.a libpthread.so)
+    if (NOT LLVM_OMP)
```

Comment:
nit: shouldn't you check `PTHREAD` too?

Reply:
Very unlikely to not have pthreads, but yeah, doesn't hurt. Added.